### PR TITLE
Add bug reporting

### PR DESCRIPTION
**Pull request by YodaLightsabr**
## Functional changes
- Adds the `REPORT_BUG` dispatch event
- Links the `REPORT_BUG` dispatch event to the bug button
- Adds the option to not copy the URL upon saving (disabled by default, only used for the `REPORT_BUG` dispatch event)

## Design changes
- Makes the images in buttons display pixelated instead of blurry

### CODE REVIEW

File: editor.css [hunk 1]
```diff
@@ -248,3 +248,7 @@ body {
 
 
 
+
+button > img {
+  image-rendering: pixelated; /* Make button icons crisp */
```
Comment (maxwofford) [inline editor.css:253]: Ooooh! TIL

File: dispatch.js [hunk 2]
```diff
@@ -120,6 +122,47 @@ const ACTIONS = {
...
+      examples: state.examples,
+      error: state.error,
+      mouseX: state.mouseX,
+      mouseY: state.mouseY,
+      engineVersion: state.engineVersion,
+      previousID: state.previousID,
+      selected_asset: state.selected_asset,
+      name: state.name,
+      lastSaved: state.lastSaved,
+    }, null, 4), 50000, '...');
```
Comment (maxwofford) [inline dispatch.js:155]: The character limit is 100k per field. Is there a reason we're truncating at 50k?

File: dispatch.js [hunk 3]
```diff
@@ -120,6 +122,47 @@ const ACTIONS = {
...
+    notification({
+      message: "Generating a bug report... (3/3)"
+    });
+    let querystring = '';
+    for (const key in report) {
+      querystring += `&prefill_${encodeURIComponent(key)}=${encodeURIComponent(report[key])}`;
+    }
+    const url = `https://airtable.com/shrpcDFA5f9wEOSIm?${querystring.substring(1)}`;
+    window.open(url, '_blank');
```
Comment (maxwofford) [inline dispatch.js:164]:
```suggestion
    const url = new URL('https://airtable.com/shrpcDFA5f9wEOSIm')
    for (const key in report) {
      url.searchParams.append(`prefill_${encodeURIComponent(key)}`, encodeURIComponent(report[key]))
    }
    window.open(url, '_blank');
```

I think this'll do the same thing, but with the built-in URL class

**Review by maxwofford — APPROVED** [review]
This looks great! Thanks for adding it. Added a couple comments, but looks good to merge. I'll bring it into my branch when I wake up

File: dispatch.js (hunk 2 above)
Comment (YodaLightsabr) [reply dispatch.js:155]: Oops! I forgot to update that.

**Comment by maxwofford** [comment]
Screw it, this looks great so I won't wait. Merging!

**Comment by YodaLightsabr** [comment]
Oops, I think the querystring thing is causing unexpected behavior. Image

**Pull request merged**